tracing/src/main/java/com/palantir/tracing/TraceLocal.java

```diff
@@ -138,6 +147,24 @@ public T remove() {
         return (T) traceLocals.remove(this);
     }
 
+    void onTraceComplete(TraceState traceState) {
```

Comment:
we take a trace state explicitly here because we're e.g. potentially going to be calling this from a detached span